scipy/sparse/lil.py

```diff
@@ -450,18 +450,32 @@ def tolil(self, copy=False):
     tolil.__doc__ = spmatrix.tolil.__doc__
 
     def tocsr(self, copy=False):
-        lst = [len(x) for x in self.rows]
-        idx_dtype = get_index_dtype(maxval=max(self.shape[1], sum(lst)))
-        indptr = np.cumsum([0] + lst, dtype=idx_dtype)
-        indices = np.empty(indptr[-1], dtype=idx_dtype)
-        data = np.empty(indptr[-1], dtype=self.dtype)
-
-        start = 0
-        for stop, indices_i, data_i in zip(indptr[1:], self.rows, self.data):
-            indices[start:stop] = indices_i
-            data[start:stop] = data_i
-            start = stop
-
+        # construct indptr array
+        n, m = self.shape
```

Comment:
this is nonstandard naming. How about `nr, nc` instead?

Other places in this file use `M, N = self.shape`, so that's acceptable too (though perhaps less obvious when reading the code).

Reply:
I updated it using `M, N = self.shape`.